jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/JDTUtils.java

```diff
@@ -625,4 +627,30 @@ public static Location toLocation(IClassFile classFile, int offset, int length)
     public static boolean hasLeadingSlash(String uriString) {
         return uriString.startsWith("/");
     }
-}
\ No newline at end of file
+    
+    /**
+     * Enumeration for the types of accessor a method can be.
+     *
+     */
+    public enum AccessorType {
+        GET, SET
+    }
+
+    /**
+     * Checks if a method is the specified type of accessor (getter or setter) of
+     * the given field. i.e. the method has the exact same name as what an accessor
+     * should have.
+     * 
+     * @param type of the accessor
+     * @param method
+     * @param field
+     * @return boolean
+     */
+    public static boolean isAccessor(AccessorType type, IMethod method, IField field) {
+        String fieldName = field.getElementName();
+        // The expected accessor name is given by the type (get/set) + the Pascal case
+        // of field name e.g. if a field is called id, the getter name is getId
+        String accessorName = type.name().toLowerCase() + fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);
```

Comment:
Looks like we are manually piecing together the getter/setter names here. 
One outlier to consider, for boolean fields often the getter is named with "is" ie.
````
boolean active;
public boolean isActive(){return this.active;}

public void setActive(boolean active){this.active = active;}

```

I'm not sure is a smarter way to determine getter/setters from JDT. There is this: 
[org/eclipse/jdt/internal/corext/codemanipulation/GetterSetterUtil.java class](https://git.eclipse.org/r/plugins/gitiles/jdt/eclipse.jdt.ui/+/2ccac2274e025ae25d67c0016439c0b7fe669f3f/org.eclipse.jdt.ui/core%20extension/org/eclipse/jdt/internal/corext/codemanipulation/GetterSetterUtil.java). There may possibly be some helper methods we can make use of.

Reply:
I looked at the GetterSetterUtil and found `getGetter` and `getSetter` very useful. I made a new `getAccessors` method in JDTUtils and removed what I had before. That way the logic is simplified in the diagnostic collectors since I no longer need to iterate over all methods to find accessors.